### PR TITLE
[C++ Worker]Add benchmark test

### PR DESCRIPTION
**Pull request by qicosmos**
## Why are these changes needed?
This pr do some benchmark for c++ worker, the test result can be used to compare with java/python worker.

## Checks

- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests

### CODE REVIEW

File: cpp/src/ray/test/benchmark/benchmark.cc [hunk 1]
```diff
@@ -0,0 +1,124 @@
...
+      .append(" average duration ")
+      .append(std::to_string(duration / num_repeats))
+      .append(" ns.");
+  RAYLOG(INFO) << out;
+
+  std::cout << out << std::endl;
+}
+
+template <typename R, typename F>
+void ExecuteLatancy(F f, std::string func_name, size_t num_repeats) {
```
Comment (SongGuyang) [inline cpp/src/ray/test/benchmark/benchmark.cc:62]: `ExecuteLatancy` ->  `RunLatencyTest`?

File: cpp/src/ray/test/benchmark/benchmark.cc [hunk 2]
```diff
@@ -0,0 +1,124 @@
...
+    return count;
+  }
+
+  int Get() const { return count; }
+};
+
+RAY_REMOTE(Counter::FactoryCreate, &Counter::Add, &Counter::Get);
+
+template <typename F>
+void Execute(F f, std::string func_name, size_t num_repeats) {
```
Comment (SongGuyang) [inline cpp/src/ray/test/benchmark/benchmark.cc:37]: Can you give an clear name of this function? I can't get the difference between `Execute` and `ExecuteLatancy`.

**Review by SongGuyang — COMMENTED** [review]
@qicosmos Can you print a human-readable benchmark report? One way is to generate a text graph. We need a clearer report to trice our performance.

File: cpp/src/ray/test/ray_remote_test.cc [hunk 3]
```diff
@@ -207,6 +214,9 @@ TEST(RayApiTest, OverloadTest) {
   EXPECT_EQ(return0, 1);
   EXPECT_EQ(return1, 2);
   EXPECT_EQ(return2, 5);
+
+  auto rt3 = ray::Task(RAY_FUNC(OverloadFunc, int, int, int)).Remote(rt1, 3, rt0);
```
Comment (kfstorm) [inline cpp/src/ray/test/ray_remote_test.cc:218]: Is this change related to benchmark?

File: cpp/BUILD.bazel [hunk 4]
```diff
@@ -223,3 +223,36 @@ cc_binary(
...
+    srcs = glob([
+        "src/ray/test/benchmark/*.cc",
+    ]),
+    copts = COPTS,
+    data = [
+        "benchmark_test.so",
+        "ray_cpp_pkg",
+    ],
+    linkstatic = True,
+    tags = ["team:serverless"],
```
Comment (kfstorm) [inline cpp/BUILD.bazel:239]: Is this required?

**Review by kfstorm — COMMENTED** [review]
Could you point out the location of the corresponding Python and Java benchmark code? And please paste the benchmark results of 3 languages in this PR.

**Comment by bveeramani** [comment]
## :bangbang: ACTION REQUIRED :bangbang:

We've switched our code formatter from YAPF to Black (see #21311).

To prevent issues with merging your code, here's what you'll need to do:

1. Install Black
```
pip install -I black==21.12b0
```

2. Format changed files with Black
```
curl -o format-changed.sh https://gist.githubusercontent.com/bveeramani/42ef0e9e387b755a8a735b084af976f2/raw/7631276790765d555c423b8db2b679fd957b984a/format-changed.sh
chmod +x ./format-changed.sh
./format-changed.sh
rm format-changed.sh
```

3. Commit your changes.
```
git add --all
git commit -m "Format Python code with Black"
```

4. Merge master into your branch.
```
git pull upstream master
```

5. Resolve merge conflicts (if necessary).

After running these steps, you'll have the updated `format.sh`.

**Comment by kfstorm** [comment]
## :bangbang: ACTION REQUIRED :bangbang:

We've updated our formatting configuration for C++ code. (see #22725)

This PR includes C++ code change. To prevent issues with merging your code, here's what you'll need to do:

1. Merge the latest changes from upstream/master branch into your branch.

```bash
git pull upstream master
git merge upstream/master
```

2. Resolve merge conflicts (if necessary).

After running these steps, you'll have the updated C++ formatting configuration.

3. Format changed files.

```bash
scripts/format.sh
```

4. Commit your changes.

```bash
git add --all
git commit -m "Format C++ code"
```
